compiler/plugins/input/StableHLO/stablehlo-iree/Conversion/StableHLOCustomCalls.cpp

```diff
@@ -0,0 +1,244 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Implements logic for lowering CHLO ops to StableHLO and Shape dialect ops,
+// taking care of CHLO's broadcasting semantics
+
+#include "llvm/ADT/STLExtras.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/ImplicitLocOpBuilder.h"
+#include "mlir/IR/TypeUtilities.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "stablehlo-iree/Conversion/Passes.h"
+#include "stablehlo-iree/Conversion/Preprocessing/Rewriters.h"
+#include "stablehlo-iree/Conversion/Rewriters.h"
+#include "stablehlo/dialect/BroadcastUtils.h"
+#include "stablehlo/dialect/StablehloOps.h"
+
+#include <algorithm>
+
+namespace mlir::iree_compiler::stablehlo {
+
+#define GEN_PASS_DEF_LEGALIZESTABLEHLOCUSTOMCALLS
+#include "stablehlo-iree/Conversion/Passes.h.inc"
+
+namespace {
+
+// Computes householder using vector `v` and a `tau` matrice
+// with the k-th element.
```

Comment:
Is there a reference we can give for the algo? (paper, textbook etc)

Reply:
I added a wikipedia reference. I could not find one good source for an implementation so I cobbled together from a few. Wikipedia's was the only one readable.